### DISCUSSION TITLE
Merge `full` prop of Modal/Drawer into `size` variants

### DISCUSSION THREAD

**Discussion by SevenOutman** [rsuite/rsuite#2454] [Ideas]
### The problem

The Modal/Drawer component is designed to have 5 different sizes - `xs` `sm` `md` `lg` and `full`. The control of size variant is now split into two separate props `size` and `full` with the behavior as described below:

- The `size` prop variates among 4 different values - `xs` `sm` `md` and `lg`, and the `full` prop is boolean
- When `full` is absent or set `false`, component's size variant regards the `size` prop
- When `full` prop is set `true`, component's size variant is set to `full` and `size` prop is ignored

Such behavior makes `size` and `full` props non-orthogonal, which means `size` prop's behavior is interfered by `full`. 

### This solution

I'd suggest deprecate the separate `full` prop and add `full` to the list of acceptable values of `size` prop. This way size variants are mutually exclusive from each other. 

Working PR #2455

**Comment by simonguo** [comment]
The intention of `full` is to allow Modal/Drawer to be full screen and fill the window. It could be summed up in the size attribute, but would need to be marked with a _breaking change_.

In addition, the current `full` prop does not really realize full screen, you can discuss with  @LeightonYao to change it to full screen.

**Reply by SevenOutman** [reply]
> but would need to be marked with a breaking change.

The deprecation would not introduce breaking change immediately. `full` prop will still be working with a deprecation warning message. Deprecated APIs are going to be completely removed then in breaking change releases.

> In addition, the current full prop does not really realize full screen

According to the design resource, “full” size does not strictly mean covering the whole viewport, but with a fixed gap (60px) to the edge. To improve flexibility we can make this gap size configurable via Less variable or CSS properties.

**Comment by simonguo** [comment]
If it just throws a deprecated warning and doesn't invalidate the `full` prop, it's fine.

About full screen is not within the scope of this modification. It is necessary to think about its scenarios from the aspects of design and use.
